spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/sanitize/KeyBlacklistSanitizeFilter.java

```diff
@@ -14,58 +14,60 @@
  * limitations under the License.
  */
 
-package org.springframework.boot.actuate.endpoint;
+package org.springframework.boot.actuate.endpoint.sanitize;
 
 import java.util.regex.Pattern;
 
 import org.springframework.util.Assert;
 
 /**
- * Strategy that should be used by endpoint implementations to sanitize potentially
- * sensitive keys.
+ * Sanitize filter operating on String keys. Matches if the given key contains any of the
+ * configured words. Supports basic regex syntax.
  *
- * @author Christian Dupuis
- * @author Toshiaki Maki
- * @author Phillip Webb
- * @author Nicolas Lejeune
- * @author Stephane Nicoll
+ * @author Piotr Betkier
  * @since 2.0.0
  */
-public class Sanitizer {
+public final class KeyBlacklistSanitizeFilter implements Sanitizer.Filter<String> {
 
 	private static final String[] REGEX_PARTS = { "*", "$", "^", "+" };
 
-	private Pattern[] keysToSanitize;
+	private final Pattern[] keysToSanitize;
 
-	public Sanitizer() {
-		this("password", "secret", "key", "token", ".*credentials.*", "vcap_services");
+	private KeyBlacklistSanitizeFilter(Pattern[] patterns) {
+		this.keysToSanitize = patterns;
 	}
 
-	public Sanitizer(String... keysToSanitize) {
-		setKeysToSanitize(keysToSanitize);
+	/**
+	 * Creates the filter based on the default blacklist.
+	 * @return a new filter instance
+	 */
+	public static KeyBlacklistSanitizeFilter matchingDefaultKeys() {
+		return KeyBlacklistSanitizeFilter.matchingKeys("password", "secret", "key",
+				"token", ".*credentials.*", "vcap_services");
 	}
 
 	/**
-	 * Keys that should be sanitized. Keys can be simple strings that the property ends
-	 * with or regular expressions.
-	 * @param keysToSanitize the keys to sanitize
+	 * Creates the filter based on the given blacklist.
+	 * @param keysToSanitize keys to match
+	 * @return a new filter instance
 	 */
-	public void setKeysToSanitize(String... keysToSanitize) {
```

Comment:
I made this class configurable by constructor only, couldn't find out why it used setters. Point out if it's a problem.